tools/tests/test_backend_datadog.py

```diff
@@ -0,0 +1,132 @@
+        self.assertEqual(query, expected_query)
+
+    def test_space_escape(self):
+        self.basic_rule["detection"]["selection"]["space-attribute"] = "with space"
+        query = self.generate_query(self.basic_rule)
+        expected_query = "@attribute:test AND @space-attribute:with?space"
+        self.assertEqual(query, expected_query)
+
+    def test_space_escape(self):
+        query = self.generate_query(
+            self.basic_rule,
+            fieldmappings={"attribute": FieldMapping("attribute", "another_attribute")},
+        )
+        expected_query = "@another_attribute:test"
+        self.assertEqual(query, expected_query)
+
+    def generate_query(
+        self, rule, backend_options=dict(), config=dict(), fieldmappings=dict()
+    ):
+        cfg = SigmaConfiguration()
+        cfg.config = config
+        cfg.fieldmappings = fieldmappings
+        backend = DatadogLogsBackend(cfg, backend_options)
+        parser = SigmaParser(rule, cfg)
+
+        return backend.generate(parser)
+
+    def test_all_sigma_rules(self):
+        """Test the Datadog backend over all the Sigma rules in the repository."""
+        verbose_report = False
+
+        skipped = 0
+        errors = 0
+        successes = 0
+        total = 0
+
+        config = SigmaConfiguration()
+        backend = DatadogLogsBackend(config)
+
+        for (dirpath, _, filenames) in os.walk("../rules"):
+            for filename in filenames:
+                if filename.endswith(".yaml") or filename.endswith(".yml"):
+                    with self.subTest(filename):
+                        rule_path = os.path.join(dirpath, filename)
+
+                        with open(rule_path, "r") as rule_file:
+                            total += 1
+                            parser = SigmaParser(yaml.safe_load(rule_file), config)
+
+                            try:
+                                query = backend.generate(parser)
+                            except NotImplementedError as err:
+                                if verbose_report:
+                                    print("[SKIPPED] {}: {}".format(rule_path, err))
+                                skipped += 1
+                            except BaseException as err:
+                                if verbose_report:
+                                    print("[FAILED] {}: {}".format(rule_path, err))
+                                errors += 1
+                            else:
+                                if verbose_report:
+                                    print("[OK] {}".format(rule_path))
+                                successes += 1
+
+        print("\n==========Statistics==========\n")
+        print(
+            "SUCCESSES: {}/{} ({:.2f}%)".format(
+                successes, total, successes / total * 100
+            )
+        )
+        print("SKIPPED: {}/{} ({:.2f}%)".format(skipped, total, skipped / total * 100))
```

Comment:
~If Sigma supports Python 2, these statistics will always show `0%` in python 2.x.~

**It looks like sigma doesn't explicitly support python2 (good!). This can be ignored then.**

Casting `total` as a float would fix this.

```python
print("SKIPPED: {}/{} ({:.2f}%)".format(skipped, total, skipped / float(total) * 100))
```

Reply:
Good to know, thanks for investigating!